File: src/jEvropa2/data/E2Data.java
package jEvropa2.data;


import java.net.URL;



/**
 *
 * @author Roman Zelenik
 */
public abstract class E2Data {
    public static final URL EMPTY_URL = null;
    protected int id;
    protected String name;

    public E2Data (String name){
        this(0, name);
    }
    
    public E2Data(int id, String name) {
        this.id = id;
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }
}

File: src/main/kt/cz/skup5/jEvropa2/data/E2Data.kt
package cz.skup5.jEvropa2.data

import java.net.URI

fun URI.isEmpty(): Boolean {
    return this.path == ""
}

/**
 * Abstract POJO E2 data object.
 *
 * @author Skup5
 */
abstract class E2Data(
        /** The unique id. */
        open val id: Int = ID_NONE,
        /** The item title/label. */
        open val name: String
) {

    fun hasId() = id > ID_NONE

    companion object {
        @JvmStatic
        val EMPTY_URI: URI = URI("")

        const val ID_NONE = -1
    }
}
